p2p/src/error.rs

```diff
@@ -14,124 +14,133 @@
 // limitations under the License.
 //
 // Author(s): A. Altonen
+use libp2p::{
+    gossipsub::error::PublishError as GossipsubPublishError,
+    swarm::{handler::ConnectionHandlerUpgrErr, DialError::*},
+};
 use thiserror::Error;
 
-// TODO: think about which errors should be returned and when
-// TODO: store peerid where appropriate!
 #[derive(Error, Debug, PartialEq, Eq)]
 pub enum ProtocolError {
+    #[error("Peer is in different network")]
     DifferentNetwork,
```

Comment:
Also useful here is to include the "difference". I imagine it's magic numbers, right? Why not include both?